src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -160,8 +225,7 @@ public override void Read(Span<byte> data)
         /// <inheritdoc />
         public override void Write(ReadOnlySpan<byte> data)
         {
-            Span<byte> dataToRead = stackalloc byte[data.Length];
-            TransferFullDuplex(data, dataToRead);
+            TransferWriteOnly(data);
```

Comment:
missing miso check here (similarly as you added for TransferFullDuplex)

Reply:
Miso isn't used for write-only though.

Reply:
Sorry, still thinking of LED strip, add a check in Read then 😄

Reply:
:) Already in TransferFullDuplex that is called by Read, do you want a check in Read as well?